builds/azure-pipelines/template-steps-build-test.yml

```diff
@@ -152,27 +151,58 @@ steps:
     checkStyleRunAnalysis: true
 
 - task: DotNetCoreCLI@2
-  displayName: '.NET Test'
+  displayName: '.NET Test on Mac (unit tests only)'
   env:
     TEST_SERVER: '$(testServer)'
     NODE_MODULES_PATH: '$(nodeModulesPath)'
     AZUREFUNCTIONS_SQLBINDINGS_TELEMETRY_OPTOUT: '1'
   inputs:
     command: test
     projects: '${{ parameters.solution }}'
-    arguments: '--configuration ${{ parameters.configuration }} ${{ parameters.testFilter }} --collect "Code Coverage" -s $(Build.SourcesDirectory)/test/coverage.runsettings'
-  condition: and(succeeded(), ne(variables['Agent.OS'], 'linux'))
+    # Disabling integration tests on macOS due to issues with running MSSQL on Docker
+    # We need to set up a self-hosted agent with Docker running by default: https://github.com/microsoft/azure-pipelines-tasks/issues/12823
+    arguments: '--configuration ${{ parameters.configuration }} --filter FullyQualifiedName!~Integration'
+  condition: and(succeeded(), eq(variables['Agent.OS'], 'Darwin'))
+
+- task: DotNetCoreCLI@2
+  displayName: '.NET Test on Windows (unit tests and CSharp integration tests)'
+  env:
+    TEST_SERVER: '$(testServer)'
+    NODE_MODULES_PATH: '$(nodeModulesPath)'
+    AZUREFUNCTIONS_SQLBINDINGS_TELEMETRY_OPTOUT: '1'
+    # Only running CSharp integration tests to get accurate code coverage numbers
+    LANGUAGES_TO_TEST: 'CSharp'
+  inputs:
+    command: test
+    projects: '${{ parameters.solution }}'
+    # Skip any non-CSharp integration tests
+    arguments: '--configuration ${{ parameters.configuration }} --filter FullyQualifiedName!~NoPropertiesThrows --collect "Code Coverage" -s $(Build.SourcesDirectory)/test/coverage.runsettings'
+  condition: and(succeededOrFailed(), eq(variables['Agent.OS'], 'Windows_NT'))
 
 - task: DotNetCoreCLI@2
-  displayName: '.NET Test on Linux'
+  displayName: '.NET Test on Windows (Java, JavaScript, OutOfProc, PowerShell, and Python integration tests)'
+  env:
+    TEST_SERVER: '$(testServer)'
+    NODE_MODULES_PATH: '$(nodeModulesPath)'
+    AZUREFUNCTIONS_SQLBINDINGS_TELEMETRY_OPTOUT: '1'
+    LANGUAGES_TO_TEST: 'Java,JavaScript,OutOfProc,PowerShell,Python'
+  inputs:
+    command: test
+    projects: '${{ parameters.solution }}'
+    # Skip any CSharp only integration tests
+    arguments: '--configuration ${{ parameters.configuration }} --filter "FullyQualifiedName~Integration & FullyQualifiedName!~AddProductsCollectorTest"'
```

Comment:
Similarly, an error is thrown if I don't skip AddProductsCollectorTest for the non-CSharp run since this test only supports CSharp.

Reply:
So we need to update the LANGUAGES_TO_TEST if we added a new language? Can you note that somewhere, may be as a comment above the SupportedLanguages enum definition?